src/clr-angular/data/datagrid/datagrid-row-detail.ts

```diff
@@ -49,14 +64,57 @@ export class ClrDatagridRowDetail<T = any> implements AfterContentInit, OnDestro
   public replacedRow = false;
 
   ngAfterContentInit() {
-    this.subscriptions.push(
-      this.expand.replace.subscribe(replaceChange => {
-        this.replacedRow = replaceChange;
-      })
-    );
+    this.subscriptions.push(this.updateViewsOnOrdersChange(), this.setReplacedOnExpandReplace());
+  }
+
+  private setReplacedOnExpandReplace(): Subscription {
+    return this.expand.replace.subscribe(replaceChange => {
+      this.replacedRow = replaceChange;
+    });
   }
 
   ngOnDestroy() {
     this.subscriptions.forEach(sub => sub.unsubscribe());
   }
+
+  private updateViewsOnOrdersChange(): Subscription {
+    return this.columnReorderService.ordersChange.subscribe((byReordering: boolean) => {
+      // we have to make sure there are views inserted in the view container
+      // before calling reorderViews method from viewManager service
+      if (byReordering && this._detailCells.length > 0) {
+        ViewManagerUtils.reorderViews(this._detailCells, this.assignRawOrders());
+        this.updateCellOrder();
+        return;
+      }
+
+      if (this.columnReorderService.orders) {
+        if (this._detailCells.length === 0) {
+          // we just need to insert the views initially when the view container is empty.
+          ViewManagerUtils.insertAllViews(this._detailCells, this.assignRawOrders(), true);
+        } else {
+          this.cells.forEach((cell, index) => {
+            if (this._detailCells.indexOf(cell._view) === -1) {
```

Comment:
nit: `Array.includes` is an option here

Reply:
Same as above, `_detailCells` is not array, but ViewContainerRef.